File: badge-management/python/badgestore.py
# badgestore.py
# This mock store simulates a database client for badges.
# It logs creation and returns a Badge object with a generated ID.

import time

class BadgeStore:
    def create(self, serial_number: str, version: str) -> dict:
        print(f"Created badge with SerialNumber: {serial_number}, Version: {version}")
        badge_id = int(time.time() * 1000)  # pseudo ID
        return {
            "id": badge_id,
            "serialNumber": serial_number,
            "version": version,
        }

def new() -> BadgeStore:
    return BadgeStore()

File: badge-management/node/badgestore.js
// badgestore.js
// This mock store simulates a database client for badges.
// It logs creation and returns a Badge object with a generated ID.

class BadgeStore {
  async Create(serialNumber, version) {
    console.log(`Created badge with SerialNumber: ${serialNumber}, Version: ${version}`);

    // Generate a pseudo ID using current timestamp
    const id = Date.now();

    return {
      id,
      serialNumber,
      version,
    };
  }
}

module.exports = { BadgeStore };
